htlcswitch/interceptable_switch.go

```diff
@@ -119,6 +124,14 @@ type FwdResolution struct {
 	// FwdActionSettle.
 	Preimage lntypes.Preimage
 
+	// OutgoingAmountMsat is the amount that is to be used for forwarding if
+	// Action is FwdActionResumeModified.
+	OutgoingAmountMsat lnwire.MilliSatoshi
+
+	// CustomRecords is the custom records that are to be used for
+	// forwarding if Action is FwdActionResumeModified.
+	CustomRecords record.CustomSet
```

Comment:
I think we might just want to accept the entire serialized blob here? Though I guess this is nice, as we can validate which records are being set to avoid override.

Reply:
Perhaps we should serialize the custom records at the point the outgoing message is serialised?